apps/wpcom-block-editor/src/wpcom/features/tracking/wpcom-autocomplete-block-search-term.js

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable import/no-extraneous-dependencies */
+
+/**
+ * External dependencies
+ */
+import { debounce, get } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { select } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import tracksRecordEvent from './track-record-event';
+
+const trackAutocompleteBlockTerm = () => {
+	const search_term = get( select( 'core/block-editor' ).getSelectedBlock(), [
+		'attributes',
+		'content',
+	] );
+
+	if ( search_term.length < 3 ) {
+		return;
+	}
+
+	const context = 'autocomplete_block';
+
+	tracksRecordEvent( 'wpcom_block_picker_search_term', {
+		search_term,
+		context,
+	} );
+
+	// Check if there are results, inspecting the DOM Tree.
+	const hasResults = !! document.querySelectorAll( '.components-autocomplete__popover' ).length;
```

Comment:
> This selector seems to broad for me to feel comfortable.

Yes, I agree.

> The Block inserter won't be the only use of this component as it's available from @wordpress/components.

I guess you mean to the Autocomplete complete. 

> Is there a way to target it more effectively?

I couldn't find a better way to target it. However, the conditions to record this event are handled before, in the `selectorHandler()` function. 
Also, let's keep in mind that [it follows a similar implementation that the `Inserter Menu` does](https://github.com/Automattic/wp-calypso/blob/master/apps/wpcom-block-editor/src/wpcom/features/tracking/wpcom-block-picker-search-term-handler.js#L51).

Reply:
I think we sholuld be safe because

* There is only a single popover `Slot` and so only 1 popover can render at a time.
* You conditionalise the code in the `selectorHandler` .